server/server.go

```diff
@@ -221,6 +223,7 @@ func (s *Server) serviceAccountsHandler(logger *log.Entry, w http.ResponseWriter
 		return
```

Comment:
If we end up in this branch, returning `404`, we won't add the header, Google does:
```
curl -v http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/xyz/ >/dev/null -H "Metadata-Flavor: Google"
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0*   Trying 169.254.169.254...
* TCP_NODELAY set
* Connected to metadata.google.internal (169.254.169.254) port 80 (#0)
> GET /computeMetadata/v1/instance/service-accounts/xyz/ HTTP/1.1
> Host: metadata.google.internal
> User-Agent: curl/7.52.1
> Accept: */*
> Metadata-Flavor: Google
>
< HTTP/1.1 404 Not Found
< Metadata-Flavor: Google
< Date: Tue, 09 Jul 2019 17:46:13 GMT
< Content-Type: text/html; charset=UTF-8
< Server: Metadata Server for VM
< Content-Length: 1610
< X-XSS-Protection: 0
< X-Frame-Options: SAMEORIGIN
```